staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -40,14 +48,26 @@ type CustomResourceDefinitionStorageStrategy struct {
 	validator       customResourceValidator
 }
 
-func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind) CustomResourceDefinitionStorageStrategy {
+func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind, crd *apiextensions.CustomResourceDefinition) CustomResourceDefinitionStorageStrategy {
+	schema := &spec.Schema{}
+	if err := apiservervalidation.ConvertToOpenAPITypes(crd, schema); err != nil {
+		utilruntime.HandleError(err)
```

Comment:
we can return an error there. Here, we can't because it is a constructor. We could add an error as return value, but that would be unidiomatic. I think moving these 7 lines (or even 8 lines, including the `validate.NewSchemaValidator(schema, nil, "", strfmt.Default)` into the caller would elegantly solve this IMO.